nixos/modules/system/boot/loader/generic-extlinux-compatible/default.nix

```diff
@@ -8,7 +8,8 @@ let
 
   timeoutStr = if blCfg.timeout == null then "-1" else toString blCfg.timeout;
 
-  builder = import ./extlinux-conf-builder.nix { pkgs = pkgs.buildPackages; };
+  builderPkgs = if (pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform) then pkgs else pkgs.buildPackages;
+  builder = import ./extlinux-conf-builder.nix { pkgs = builderPkgs; };
```

Comment:
I wolud just do ```pkgs = pkgs.buildPackages```. On hostPlatform == buildPlatform, it will be the same as pkgs. This would remove the above condition.

Reply:
The important case I want to address is `hostPlatform != buildPlatform`.
If I just pass `pkgs = pkgs.buildPackages`, it won't work.

Just to clarify, you suggest to remove the condition and use
```
builder = import ./extlinux-conf-builder.nix { pkgs = pkgs.buildPackages; };
```

If so, that fails for me.

Reply:
Oh ok! I see, I was thinking these would be run on the building machine. But it looks like the install bootloader is run on the target machine. That makes sense now that I think about it. I think a few more of these are messed up. These were done by me in this commit:

https://github.com/NixOS/nixpkgs/commit/35af6e36057cafbb30df684326803e9e54bb377e

Could you see if reverting that fixes the issue? I would prefer to reduce the number of if statements we use here.

Reply:
I think we need to revert that commit for all of them besides the sd-* ones. Those /are/ run on the build machine, but installBootLoader is not

Reply:
Pushed the revert to a second branch to keep it clean
#62333

Tests work out well so far, but the sd-image does not build for me (uboot won't compile), so that part is untested. Will close this PR when #62333 is accepted.